waves/src/App.tsx

```diff
@@ -167,29 +203,49 @@ function App() {
         };
     }, [rateService]);
 
-
     useEffect(() => {
         getWalletStatus().then((wallet_status) => {
             if (wallet_status.exists) {
                 dispatch({
-                    type: "UpdateWallet",
+                    type: "UpdateWalletStatus",
                     value: {
                         exists: wallet_status.exists,
                         loaded: wallet_status.loaded,
-                        btcBalance: 0,
-                        usdtBalance: 0,
                     },
                 });
             } // by default `wallet.exists` is set to false, hence no need to handle
         });
     }, []);
 
+    useEffect(() => {
+        let updateBalanceInterval = setInterval(() => {
```

Comment:
There is a library that provides a hook for this if you don't want to roll your own: https://usehooks-typescript.com/react-hook/use-interval

Reply:
I saw in the the next patch that you are actually modifying this further. Might be a good idea to use the hook from the library then.

Reply:
So much nicer with this hook :)